boot/bootutil/include/bootutil/bootutil.h

```diff
@@ -65,6 +65,38 @@ int boot_swap_type(void);
 int boot_set_pending(int permanent);
 int boot_set_confirmed(void);
 
+#define MCUBOOT_REQ_FLASH_MAP_SIZE          0
+#define MCUBOOT_REQ_FLASH_MAP_INFO          1
+
+struct mcuboot_api_itf {
+    uint32_t mcuboot_api_magic;
+    uint32_t mcuboot_version;
+    int (*mcuboot_ioctl)(int req, void *data, size_t len);
+};
+
+extern const struct mcuboot_api_itf mcuboot_api_vt;
+extern struct mcuboot_api_itf *p_mcuboot_api_vt;
+
+#if defined(__arm__)
+    #define MCUBOOT_API_INIT_(m)                                            \
+        extern const struct mcuboot_api_itf mcuboot_api_vt;                 \
+        asm("mov r4, %0\n\t"                                                \
+            "mov r5, %1"  :: "I" (m), "r" (&mcuboot_api_vt) : "r4", "r5");
```

Comment:
The thing is the "clobbers" won't be used to prep and post the asm, but the compiler would be free to use those registers after, including for the call following it. It turns out that we're ok here because the call has no args, so the compiler feels no need to put anything else in the registers.

But, I'd be more comfortable if it was explicit, either putting the jump in the asm itself (along with the arguments), or just using a regular call with arguments.